src/client/websocket/handlers/INTERACTION_CREATE.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const { Events, InteractionTypes } = require('../../../util/Constants');
+let Structures;
+
+module.exports = (client, { d: data }) => {
+  if (data.type === InteractionTypes.APPLICATION_COMMAND) {
+    if (!Structures) Structures = require('../../../util/Structures');
+    const CommandInteraction = Structures.get('CommandInteraction');
+
+    const interaction = new CommandInteraction(client, data);
+
+    /**
+     * Emitted when an interaction is created.
+     * @event Client#interaction
+     * @param {Interaction} interaction The interaction which was created
+     */
+    client.emit(Events.INTERACTION_CREATE, interaction);
```

Comment:
```suggestion
    const Interaction = new CommandInteraction(client, data);

    /**
     * Emitted when an interaction is created.
     * @event Client#interaction
     * @param {Interaction} interaction The interaction which was created
     */
    client.emit(Events.INTERACTION_CREATE, Interaction);
```
Variable `interaction` is not declared with an uppercase `i` making the parameter type in the JSDoc appear as `any`.

Reply:
the type in the JSDoc should really be `CommandInteraction` and the variable name should stay as it is

Reply:
It won't always be a `CommandInteraction` though, so it should stay as is, e.g. `ButtonInteraction` will be coming soon

Reply:
I think that's better yea, edited.

Reply:
Basically what @ckohen just said. There will be more types of interactions in the future, so `Interaction` is more fitting

Reply:
@ckohen Both would be different interaction types though, so we could add them as
`@param {CommandInteraction | ButtonInteraction} interaction`
in the future when its out. Or just wait and add both at the same time if that's what you're referring to.

Reply:
All Interactions will extend the Interaction class, this is valid jsdoc and is how other extended classes (e.g. channelCreate event) are documented

Reply:
Ah I see, then seems good as it is.